blockchain/blockdao/blockindexer_withstart_group.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package blockdao
+
+import (
+	"context"
+	"errors"
+
+	"github.com/iotexproject/iotex-core/blockchain/block"
+)
+
+type (
+	// BlockIndexerWithStart defines an interface to accept block to build index from a start height
+	BlockIndexerWithStart interface {
+		BlockIndexer
+		// StartHeight returns the start height of the indexer
+		StartHeight() uint64
+	}
+
+	// BlockIndexerWithStartGroup defines a group of block indexers with start height
+	BlockIndexerWithStartGroup struct {
+		indexers      []BlockIndexerWithStart
+		tipHeightFunc func() (uint64, error)
+	}
+)
+
+// NewBlockIndexerWithStartGroup creates a new block indexer group
+func NewBlockIndexerWithStartGroup(indexers ...BlockIndexerWithStart) *BlockIndexerWithStartGroup {
+	return &BlockIndexerWithStartGroup{
+		indexers: indexers,
+	}
+}
+
+// SetTipHeightFunc sets the max valid height getter
+// it must be set before Start
+func (g *BlockIndexerWithStartGroup) SetTipHeightFunc(f func() (uint64, error)) {
+	g.tipHeightFunc = f
+}
+
+// Start starts the block indexer group
+func (g *BlockIndexerWithStartGroup) Start(ctx context.Context) error {
+	if g.tipHeightFunc == nil {
+		return errors.New("tipHeightFunc is not set")
+	}
+	for _, indexer := range g.indexers {
+		if err := indexer.Start(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
i think we don't need to change `Start()`, but need to change `dao.checkIndexer`

inside there, if it's a `IndexerWithStart`, then we only do the check indexer for blocks that's > each indexer's `StartHeight`